src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.IO;
+using Newtonsoft.Json;
+using Microsoft.AspNetCore.Http;
+using System.Net.Http.Headers;
+
+namespace Microsoft.R.Host.Broker.RemoteUri {
+    public class RemoteUriHelper {
```

Comment:
MVC stuff will just get in the way here. This just needs to forward HTTP requests, maintaining them as much intact as possible, so the fewer steps of the pipeline they have to go through before being handled, the better. MVC action result handling is also less than helpful for this purpose.